### ISSUE TITLE
Improvement for temporary messages

### ISSUE THREAD

**Issue by sebastianbauer** [wireapp/wire#44]
Temporary messages are temporary per client. If you have 10 clients added, you will receive temporary messages 10 times with their own counters. So I can open the same messages even after a very long time.

After the first successful transfer it should be dropped.

**Comment by david-wire** [comment]
Thanks for the comment. You can have a maximum of 8 devices registered to your Wire account.
The countdown starts from the moment the conversation has a "focus" on a particular timed message (on a given device). As soon as the countdown ends, the same timed message should disappear on your other devices too.

Please give us some additional information if this is not what you have experienced.

**Comment by sebastianbauer** [comment]
So it's 8 devices. Yes, that's not what I experienced, but maybe another use case. I had a chat on device A where the messages disappeared as expected. Way after this I ran Wire on device B which was offline in the meantime. The same messages appeared again there with the initial timer.

**Comment by david-wire** [comment]
What were devices A & B exactly (device type)? I could not reproduce this with MacOS as device A and an iOS device as device B.

**Comment by sebastianbauer** [comment]
I used Linux and Android on a regular basis and opened a Windows client after a while and got that behaviour.

**Comment by sebastianbauer** [comment]
Seems I can be more precise how to reproduce it: If the second device is completely offline, and gets online after the temporary messages expired, then all seems to be fine. But if the second device was online and fetched the messages, they will stay there "for ever" until I focus them.

Actually, I would expect they will also disappear.

**Comment by david-wire** [comment]
Sorry for the delay in responding.

> But if the second device was online and fetched the messages, they will stay there "for ever" until I focus them.

Meaning that on the first device, the timer is still running?

**Comment by sebastianbauer** [comment]
Yes, in this scenario the timer was still running on device 1 as device 2 was online and received the messages. The timer starts on device 2 on focus, even when the message was already expired on device 1.

**Comment by wmengli** [comment]
Hi @sebastianbauer 
Thanks for the report and sorry for taking so long to reply. We tried to reproduce it and only reproducible once with one of the test phone we have. We have opened a bug for internal investigation.

**Comment by sebastianbauer** [comment]
Seems a feature nobody uses (on multiple devices).

I have one chat where I only use temporary messages. I'm using this chat almost only on my Linux machine. Just opened this chat on my Windows machine and I get all those messages back from June.

They should dissapear after 5 minutes. But to make it even worse: Only the timer of those messages was activated, that were loaded. Since you do some lazy loading I can scroll back and get older and older messages. All with a 5 minutes timer. Really uncool.

**Comment by cityr0ler** [comment]
Hi @sebastianbauer,
thanks for your report and I am sorry that you don't have the greatest user experience at the moment.
We could reproduce this with Android but since in your case the is no Android involved it is different. By design a timed message should not be sent to your own other devices, only to the ones of your chat partners. Do your Linux machine and Windows machine use the same account?

**Comment by sebastianbauer** [comment]
@cityr0ler I use the same account on all my devices. Just double checked.

Any chance to start Wire with some debug logging or something?

**Comment by wmengli** [comment]
@sebastianbauer , once again we reproduced this in-house and got the logs. We don't know the root cause yet and won't be able to fix it soon as we are working on other higher priority items. Very sorry about this.
